### DISCUSSION TITLE
Guide to working with multiple files (e.g., one chapter per file)?

### DISCUSSION THREAD

**Discussion by davidleejy** [typst/typst#2201] [Q&A] [answered]
Greetings fellow Typst users,

Having experienced how LaTeX allows the combining of multiple separate files together to form a single document, I was wondering if there might be a guide for how to accomplish this in Typst that mentions how to get equation numbering to flow right across the separate files? I believe that there would be interest from authors who are composing long, multi-chapter documents (e.g., books, theses) in figuring this out.

Casually played around with the `include` command to copy content from separate files into a single one (e.g., copy content from `chpt1.typ`, `chpt2.typ`, etc into `main.typ`). However, I have yet to figure out how to get equation numbering to flow correctly across separate files 😦

Would be grateful for help from anyone who has experienced success in doing this 😊

**Comment by laurmaedje** [comment, marked as answer]
That's a good idea for a guide! 

With the equation numbering, I wonder what's going wrong because it shouldn't be affected by how you split up your content. If you `#set math.equation(numbering: "(1)")` in main.typ all included files (after the set rule) will also be affected.

**Reply by davidleejy** [reply]
> If you #set math.equation(numbering: "(1)") in main.typ all included files (after the set rule) will also be affected.

Ah thank you, this works 🙌🏻. Had a nagging suspicion that the project I was using as a sandbox for learning Typst was having issues with equation numbering across multiple files due to it bearing a mildly tangled mess of templates and imports 😅.

In case anyone is wondering how set up a multi-file project with equation numbering and referencing working correctly across files, I've created a basic working example containing three files (`main.typ`, `chpt1.typ`, `chpt2.typ`) that I hope is helpful to kickstart your project:

<img width="474" alt="Screenshot 2023-09-21 at 7 19 12 PM" src="https://github.com/typst/typst/assets/1690072/88d99ae0-f427-4bce-8c27-883d0548eb10">

In `main.typ`
```Typst
#set heading(numbering: "1.1")
#set math.equation(numbering: "(1)" )

#include "chpt1.typ"

#include "chpt2.typ"
```

In `chpt1.typ`
```Typst
= Chapter 1 Main Header

== Chapter 1 subheader 1

lorem ipsum lorem ipsum

$ y = 2x $ <eq-test>

Referencing @eq-test-1.

== Chapter 1 subheader 2

lorem ipsum lorem ipsum

$ y = 6x $ <eq-test-1>

Referencing @eq-test and @eq-range-small.
```

In `chpt2.typ`
```Typst
= Chapter 2 Main Header

== Chapter 2 subheader 1

lorem ipsum lorem ipsum

$ | x - c | < delta $ <eq-interval-small>

Referencing @eq-interval-small.

== Chapter 2 subheader 2

lorem ipsum lorem ipsum

$ | f(x) - f(c) | < epsilon $ <eq-range-small>

Referencing @eq-range-small and @eq-test-1.
```

**Reply by RDeRenzi** [reply]
For more complex equation numbering schemes, like

> 

= Chapter 1 
== Section 1.1
Eq. 1.1.1
Eq.1.1.2
...
== Section 1.2
Eq. 1.2.1

= Chapter 2
== Section 2.1
Eq. 2.1.1

see e.g. [bookly](https://typst.app/universe/package/bookly/)
